kmir/Makefile

```diff
@@ -121,11 +121,24 @@ check-black: poetry-install
 
 # Kompilation
 
-kbuild-%:
+kbuild-%: check-k-version
 	$(POETRY_RUN) kbuild kompile $*
 
 kwhich-%:
 	$(POETRY_RUN) kbuild which $*
 
+check-k-version:
+	./check_k_version.sh
+
 .PHONY: kbuild-*
 
+# Nix (doesn't perform version check)
+
+.PHONY: nix-build-backends
+
+nix-build-backends: clean nix-kbuild-llvm nix-kbuild-haskell
+
+nix-kbuild-%:
+	$(POETRY_RUN) kbuild kompile $*
+
+.PHONY: nix-kbuild-*
```

Comment:
I'm happy with these changes in principle but I'd like to understand _why_ the Nix version check doesn't work; that might well be a bug we should fix in upstream K.

Reply:
It could in theory be a different version, yes. `kup` provides the `--override PACKAGE VERSION` flag to change the version of a flake input in development. This might be something you want to allow, as a sort of moral equivalent to the user changing the script in development to get a different K version, or it might be something you want to forbid.

I'm happy either way; whatever option you prefer for your development and usage ergonomics is fine, but I just wanted to check that there wasn't a _bug_ in the Nix version of K that meant you couldn't get the version properly.